rmc/utils/constraint.py

```diff
@@ -955,214 +933,235 @@ def get_rescue_transcripts_and_create_no_breaks_ht(overwrite: bool) -> None:
+        Assumes there is a folder for each search round run, regardless of whether there were breaks discovered
 
-    :param ht: Input Table.
-    :param section_num: Transcript section number (e.g., 1 for first section, 2 for second, 3 for third, etc.).
-    :param section_type: Transcript section type. Must be one of 'first', 'middle', or 'end'.
-    :param indices: List of indices pointing to breakpoints.
-    :return: Table containing transcript and new section obs, exp, and chi square annotations.
+    :param is_rescue: Whether to operate on searches in rescue pathway.
+    :return: Sorted list of round numbers.
     """
-    assert section_type in {
-        "first",
-        "middle",
-        "last",
-    }, "section_type must be one of 'first', 'middle', 'last'!"
-
-    logger.info("Getting info for section of transcript between two breakpoints...")
-    if section_type == "first":
-        logger.info(
-            "Getting info for first section of transcript (up to and including smallest breakpoint pos)..."
+    single_search_round_nums = get_break_search_round_nums(
+        single_search_bucket_path(is_rescue=is_rescue)
+    )
+    simul_search_round_nums = get_break_search_round_nums(
+        simul_search_bucket_path(is_rescue=is_rescue)
+    )
+    logger.info(
+        "Single search round numbers: %s\nSimultaneous search round numbers: %s",
+        ",".join(map(str, single_search_round_nums)),
+        ",".join(map(str, simul_search_round_nums)),
+    )
+    if single_search_round_nums != list(range(1, max(single_search_round_nums))):
+        raise DataException(
+            "Single search round numbers are not consecutive and increasing from 1, please double-check!"
         )
-        ht = ht.filter(ht.locus.position <= ht.break_pos[0])
-        ht = ht.annotate(
-            # Start position is transcript start if this is the first section
-            section_start_pos=ht.start_pos,
-            section_end_pos=ht.break_pos[0],
+    if simul_search_round_nums != list(range(1, max(simul_search_round_nums))):
+        raise DataException(
+            "Simultaneous search round numbers are not consecutive and increasing from 1, please double-check!"
         )
-    elif section_type == "middle":
-        ht = ht.filter(
-            (ht.locus.position > ht.break_pos[indices[0]])
-            & (ht.locus.position <= ht.break_pos[indices[1]])
+    if single_search_round_nums != simul_search_round_nums:
+        raise DataException(
+            "Round numbers from single and simultaneous searches do not match, please double-check!"
         )
-
-        # Add section start/end position annotations
-        ht = ht.annotate(
-            # Add 1 to break_pos[indices[0]] since it isn't actually included in the region
-            section_start_pos=ht.break_pos[indices[0]] + 1,
-            section_end_pos=ht.break_pos[indices[1]],
+    if len(single_search_round_nums) == 0:
```

Comment:
should this check be first? the check against
```
list(range(1, max(single_search_round_nums)))
```
would throw an error if `single_search_round_nums` is empty, right?